__tests__/app.test.js

```diff
@@ -71,6 +71,95 @@ describe("/api/reviews", () => {
+      .get("/api/reviews")
+      .expect(200)
+      .then(({ body: { reviews } }) => {
+        expect(reviews).toBeSorted({ descending: true });
+      });
+  });
+  describe("queries", () => {
+    test("reviews are by sorted in ascending order when queried", () => {
+      return request(app)
+        .get("/api/reviews?order=asc")
+        .expect(200)
+        .then(({ body: { reviews } }) => {
+          expect(reviews).toBeSorted({ descending: false });
+        });
+    });
+    test("reviews are by orderd by date by default", () => {
+      return request(app)
+        .get("/api/reviews")
+        .expect(200)
+        .then(({ body: { reviews } }) => {
+          expect(reviews).toBeSorted({ descending: true, key: "created_at" });
+        });
+    });
+    test("reviews are ordered by title when queried", () => {
+      return request(app)
+        .get("/api/reviews?sort_by=title")
+        .expect(200)
+        .then(({ body: { reviews } }) => {
+          expect(reviews).toBeSorted({ descending: true, key: "title" });
+        });
+    });
+    test("reviews are ordered by title ascendingly when queried", () => {
+      return request(app)
+        .get("/api/reviews?sort_by=title&order=asc")
+        .expect(200)
+        .then(({ body: { reviews } }) => {
+          expect(reviews).toBeSorted({ descending: false, key: "title" });
+        });
+    });
+    test("reviews are filterd by category dexterity when queried", () => {
+      const expected = {
+        title: "Jenga",
+        designer: "Leslie Scott",
+        owner: "philippaclaire9",
+        review_id: 2,
+        review_img_url:
+          "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+        category: "dexterity",
+        votes: 5,
+        comment_count: "3",
+        created_at: "2021-01-18T10:01:41.251Z",
+      };
+      return request(app)
+        .get("/api/reviews?category=dexterity")
+        .expect(200)
+        .then(({ body: { reviews } }) => {
+          expect(reviews.length).toBe(1);
+          expect(reviews[0]).toEqual(expected);
+        });
+    });
+  });
+  describe("errors", () => {
+    test("invalid order results in code 400 - msg: bad request", () => {
+      return request(app)
+        .get("/api/reviews?order=invalid")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("bad request");
+        });
+    });
+    test("invalid sorter results in code 400 - msg: bad request", () => {
+      return request(app)
+        .get("/api/reviews?sort_by=invalid")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("bad request");
+        });
+    });
+    test("invalid category results in code 404 - msg: category not found", () => {
+      return request(app)
+        .get("/api/reviews?category=invalid")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("category not found");
+        });
+    });
+  });
 });
```

Comment:
excellent tests so far; one more to consider: how would you want your app to respond to a valid **category** that has no associated reviews?